plugin/builtin/prefixtransformer/PrefixTransformer.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate pluginator
+package main
+
+import (
+	"errors"
+	"sigs.k8s.io/kustomize/api/filters/prefix"
+
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// Add the given prefix to the field
+type plugin struct {
+	Prefix     string        `json:"prefix,omitempty" yaml:"prefix,omitempty"`
+	FieldSpecs types.FsSlice `json:"fieldSpecs,omitempty" yaml:"fieldSpecs,omitempty"`
+}
+
+//noinspection GoUnusedGlobalVariable
+var KustomizePlugin plugin
+
+var prefixFieldSpecsToSkip = types.FsSlice{
+	{Gvk: resid.Gvk{Kind: "CustomResourceDefinition"}},
+	{Gvk: resid.Gvk{Group: "apiregistration.k8s.io", Kind: "APIService"}},
+	{Gvk: resid.Gvk{Kind: "Namespace"}},
+}
+
+func (p *plugin) Config(
+	_ *resmap.PluginHelpers, c []byte) (err error) {
+	p.Prefix = ""
+	p.FieldSpecs = nil
+	err = yaml.Unmarshal(c, p)
+	if err != nil {
+		return
+	}
+	if p.FieldSpecs == nil {
+		return errors.New("fieldSpecs is not expected to be nil")
+	}
+	return
+}
+
+func (p *plugin) Transform(m resmap.ResMap) error {
+	// Even if the Prefix is empty we want to proceed with the
+	// transformation. This allows to add contextual information
+	// to the resources (AddNamePrefix).
+	for _, r := range m.Resources() {
+		// TODO: move this test into the filter (i.e. make a better filter)
+		if p.shouldSkip(r.OrgId()) {
+			continue
+		}
+		id := r.OrgId()
+		// current default configuration contains
+		// only one entry: "metadata/name" with no GVK
+		for _, fs := range p.FieldSpecs {
+			// TODO: this is redundant to filter (but needed for now)
+			if !id.IsSelected(&fs.Gvk) {
+				continue
+			}
+			// TODO: move this test into the filter.
+			if fs.Path == "metadata/name" {
```

Comment:
Suggestion: put "metadata/name" as a constant at the top of this file.

Reply:
Once the plugins are generated these are all located in the `main` package. If I move this to a constant at the top of the file the name would need to differ between the `PrefixTransformer` and `SuffixTransformer` (as the plugins generated are both in `main`, thus the variables are too).

I think that a workahound like naming the variable `metadataNamePathPrefix` or `prefixMetadataNamePath` and the equivalent of `metadataNamePathSuffix` or `suffixMetadataNamePath` is more of a code smell than this.

What do you think? I don't think I'm the first person who runs into the fact that everything is co-located in `main` after generation 😄

Reply:
> I think that a workahound like naming the variable metadataNamePathPrefix or prefixMetadataNamePath and the equivalent of metadataNamePathSuffix or suffixMetadataNamePath is more of a code smell than this.

Yeah, agreed. Thank you for the explanation, we can leave it as it is then.